src/WaitForReact.test.js

```diff
@@ -0,0 +1,301 @@
+    ));
+
+    await new Promise((resolve) => setTimeout(resolve, 550));
+
+    const calledLowerThanZero = defaultChildren.mock.calls.some((call) => call[0].progress < 0);
+    const calledHigherThanOrEqualToOne = defaultChildren.mock.calls.some((call) => call[0].progress >= 1);
+
+    expect(calledLowerThanZero).toBe(false);
+    expect(calledHigherThanOrEqualToOne).toBe(false);
+    expect(defaultChildren).toHaveBeenCalledWith({ progress: 0.95, error: undefined });
+});
+
+describe('props change', () => {
+    it('should allow changing progressInterval', async () => {
+        const promise = new Promise(() => {});
+        const { rerender } = render((
+            <Tree
+                maxProgressBeforeInteractive={ 0.2 }
+                progressInterval={ 50 }
+                promise={ promise } />
+        ));
+
+        rerender((
+            <Tree
+                maxProgressBeforeInteractive={ 0.2 }
+                progressInterval={ 700 }
+                promise={ promise } />
+        ));
+
+        await new Promise((resolve) => setTimeout(resolve, 600));
+
+        expect(defaultChildren).toHaveBeenCalled();
+
+        const called = defaultChildren.mock.calls.some((call) => call[0].progress > 0.2 && call[0].progress < 1);
+
+        expect(called).toBe(false);
+    });
+
+    it('should allow changing progressDecay', async () => {
+        const promise = new Promise(() => {});
+        const { rerender } = render((
+            <Tree
+                maxProgressBeforeInteractive={ 0.2 }
+                promise={ promise } />
+        ));
+
+        await new Promise((resolve) => setTimeout(resolve, 50));
+
+        rerender((
+            <Tree
+                maxProgressBeforeInteractive={ 0.2 }
+                progressDecay="() => 0.9"
+                promise={ promise } />
+        ));
+
+        await new Promise((resolve) => setTimeout(resolve, 200));
+
+        expect(defaultChildren).toHaveBeenCalled();
+
+        const called = defaultChildren.mock.calls.some((call) => call[0].progress > 0.9 && call[0].progress < 1);
+
+        expect(called).toBe(true);
+    });
+
+    it('should allow changing promise', async () => {
+        const { rerender } = render(<Tree promise={ new Promise(() => {}) } />);
+
+        await new Promise((resolve) => setTimeout(resolve, 250));
+
+        rerender(<Tree promise={ Promise.resolve() } />);
+
+        await new Promise((resolve) => setTimeout(resolve, 250));
+
+        expect(defaultChildren).toHaveBeenCalled();
+        expect(defaultChildren.mock.calls.length).toBeGreaterThanOrEqual(3);
+        expect(defaultChildren.mock.calls[defaultChildren.mock.calls.length - 1][0]).toEqual({ progress: 1, error: undefined });
+    });
+});
+
+describe('SSR', () => {
+    let windowSpy;
+
+    beforeEach(() => {
+        windowSpy = jest.spyOn(global, 'window', 'get');
+    });
+
+    afterEach(() => {
+        windowSpy.mockRestore();
+        delete window.__WAIT_FOR_IT__;
```

Comment:
See comment above if you accept the suggestion.

```suggestion
        delete window.__REACT_WAIT_FOR_IT__;
```